GetStream/stream-chat-android
# [ISSUE-4638] Quoted messages db fix

### 🎯 Goal

Fixing pagination problems due to the wrong insertion of replyTo messages inside the database.

### 🛠 Implementation details

Another Dao was created, exclusive for quoted messages. Then the quoted messages are requested only when inside a message that has a quoted message and not together with all the normal messages. 

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/10619102/214347301-8cb9e63f-598e-4cc3-9f05-a9eeff379f1a.mov" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/10619102/214347261-4d8cb493-90cb-47a8-a3a3-512ad287da79.mov" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing
- Reply to a old message and let it be the last message of the channel.
- Clear the database.
- Load the first page of messages of a channel. **Only the first page!**
- Disable internet
- Go to the end of a page. 

The last message **should not** be the quoted message. Like you see in the before and after videos.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![giphy (1)](https://user-images.githubusercontent.com/10619102/214306923-8ae7165c-acc1-447a-bb62-3d00a9e2ad42.gif)
_"Quotes"_
